crates/primitives/src/snapshot/segment.rs

```diff
@@ -81,17 +93,26 @@ impl SnapshotSegment {
     }
 
     /// Takes a filename and parses the [`SnapshotSegment`] and its inclusive range.
-    pub fn parse_filename(name: &str) -> Option<(Self, RangeInclusive<BlockNumber>)> {
+    pub fn parse_filename(
+        name: &str,
+    ) -> Option<(Self, RangeInclusive<BlockNumber>, RangeInclusive<TxNumber>)> {
         let parts: Vec<&str> = name.split('_').collect();
```

Comment:
we should be able to do this without an alloc.
by having options that we set while iterating over the str and count all the items